scenarios/kubernetes_e2e.py

```diff
@@ -44,49 +45,169 @@ def check(*cmd):
-    if not os.path.isdir(artifacts):
-        os.makedirs(artifacts)
+class LocalMode(object):
+    """Runs e2e tests by calling e2e-runner.sh."""
+    def __init__(self, workspace):
+        self.workspace = workspace
+        self.env = []
+        self.env_files = []
+        self.add_environment(
+            'HOME=%s' % workspace,
+            'WORKSPACE=%s' % workspace,
+            'PATH=%s' % os.getenv('PATH'),
+        )
+
+    @staticmethod
+    def parse_env(env):
+        """Returns (FOO, BAR=MORE) for FOO=BAR=MORE."""
+        return env.split('=', 1)
+
+    def add_environment(self, *envs):
+        """Adds FOO=BAR to the list of environment overrides."""
+        self.env.extend(self.parse_env(e) for e in envs)
+
+    def add_files(self, env_files):
+        """Reads all FOO=BAR lines from each path in env_files seq."""
+        for env_file in env_files:
+            with open(test_infra(env_file)) as fp:
+                for line in fp:
+                    line = line.rstrip()
+                    if not line or line.startswith('#'):
+                        continue
+                    self.env_files.append(self.parse_env(line))
+
+    def add_gce_ssh(self, priv, pub):
+        """Copies priv, pub keys to $WORKSPACE/.ssh."""
+        gce_ssh = '%s/.ssh/google_compute_engine' % self.workspace
+        gce_pub = '%s/.ssh/google_compute_engine.pub' % self.workspace
+        shutil.copy(priv, gce_ssh)
+        shutil.copy(pub, gce_pub)
+        self.add_environment(
+            'JENKINS_GCE_SSH_PRIVATE_KEY_FILE=%s' % gce_ssh,
+            'JENKINS_GCE_SSH_PUBLIC_KEY_FILE=%s' % gce_pub,
+        )
+
+    def add_service_account(self, path):
+        """Sets GOOGLE_APPLICATION_CREDENTIALS to path."""
+        self.add_environment('GOOGLE_APPLICATION_CREDENTIALS=%s' % path)
+
+    @property
+    def runner(self):
+        """Finds the best version of e2e-runner.sh."""
+        options = ['e2e-runner.sh', test_infra('jenkins/e2e-image/e2e-runner.sh')]
+        for path in options:
+            if os.path.isfile(path):
+                return path
+        raise ValueError('Cannot find e2e-runner at any of %s' % ', '.join(options))
+
+
+    def install_prerequisites(self):
+        """Copies upload-to-gcs and kubetest if needed."""
+        parent = os.path.dirname(self.runner)
+        if not os.path.isfile(os.path.join(parent, 'upload-to-gcs.sh')):
+            shutil.copy(
+                test_infra('../kubernetes/hack/jenkins/upload-to-gcs.sh'),
+                os.path.join(parent, 'upload-to-gcs.sh'))
+        if not os.path.isfile(os.path.join(parent, 'kubetest')):
+            check('go', 'install', 'k8s.io/test-infra/kubetest')
+            shutil.copy(
+                os.path.expandvars('${GOPATH}/bin/kubetest'),
+                os.path.join(parent, 'kubetest'))
+
+    def start(self):
+        """Runs e2e-runner.sh after setting env and installing prereqs."""
+        env = {}
+        env.update(self.env_files)
```

Comment:
Oh... it's parsed above.. nvm.. I was thinking python has a native env-file handler or something..